Python/Product/VSInterpreters/Interpreter/PythonWorkspaceContext.cs

```diff
@@ -63,6 +62,19 @@ class PythonWorkspaceContext : IPythonWorkspaceContext {
             _optionsService = optionsService ?? throw new ArgumentNullException(nameof(optionsService));
             _registryService = registryService ?? throw new ArgumentNullException(nameof(registryService));
             _workspaceSettingsMgr = _workspace.GetSettingsManager();
+
+            // Initialization in 2 phases (Constructor + Initialize) is needed to
+            // break a circular dependency.
+            // We create a partially initialized object that can be used by
+            // WorkspaceInterpreterFactoryProvider to discover the interpreters
+            // in this workspace (it needs the interpreter setting to do that).
+            // Once that is done, the IPythonInterpreterFactory on this object
+            // can be resolved in Initialize.
+            _interpreter = ReadInterpreterSetting();
```

Comment:
looks like ReadInterpreterSetting() can throw ArgumentNullException if you follow its calls. should we add a catch?

Reply:
i guess we check for null workspace above this so should be ok

Reply:
exactly